app/views/metadata.js

```diff
@@ -380,6 +389,91 @@ module.exports = Backbone.View.extend({
     return false;
   },
 
+  toggleImageDialog: function(e) {
+    if (e === undefined) {
+      $('.meta .dialog').remove();
+      return false;
+    }
+
+    var self = this;
+    var $parent = $(e.target).parent();
+    var $lastChild = $parent.children().last();
+
+    //Toggle Dialog On/Off
+    if ($lastChild.hasClass('dialog')) {
+      $lastChild.remove();
+    } else {
+
+      //Create the dialog element
+      var $dialog = $('<div></div>');
+      $parent.append($dialog);
+      $dialog.addClass('dialog media');
+
+      //Populate the dialog element
+      tmpl = _(templates.dialogs.media).template();
+      $dialog.append(tmpl({
+        description: t('dialogs.media.description', {
+          input: '<input data-element="upload" class="upload" type="file" />'
+        }),
+        assetsDirectory: (self.view.toolbar.media && self.view.toolbar.media.length) ? true : false,
+        writable: self.model.get('writable'),
+        altText: false
+      }));
+
+      $media = $dialog.find('ul[data-element="media"]').first();
+      $input = $dialog.find('input[data-element="upload"]').first();
+      $input.change(self.fileInput);
+
+      if (self.view.toolbar.media && self.view.toolbar.media.length) util.renderMedia(self.view.toolbar.media, $media, this);
+    }
+    return false;
+  },
+
+  fileInput: function(e) {
+    var $dialog = $(e.target).closest('div.dialog.media');
+    var self = this;
+    upload.fileSelect(e, function(e, file, content) {
+      self.trigger('updateImageInsert', e, file, content, self, $dialog);
+    });
+
+    return false;
+  },
+
+  updateInput: function(e, file, content) {
+    var $dialog = $(e.target, this.$el).closest(".dialog");
+    var $input = $dialog.parent().children("input");
+    var self = this;
+
+    if (this.queue) {
+      var userDefinedPath = $('input[name="url"]').val();
+
+      var onSuccess = function(model, res, options) {
```

Comment:
I would guess that this is where it's failing to set the save state, and causing the spinner to endlessly spin.